File: src/javaasync/escalation/EscalationReport.java
package javaasync.escalation;

/**
 * Escalation report.
 */
public abstract class EscalationReport
{

    /*
     * Flag to see if recovered.
     */
    protected boolean recovered = false;

    /**
     * Handle escalation.
     */
    public abstract void handle();

    /**
     * Get reason.
     *
     * @return reason in textual form.
     */
    public abstract String reason();
}

File: src/javaasync/escalation/EscalationReport.kt
package javaasync.escalation

/**
 * Escalation report.
 */
abstract class EscalationReport {

    /*
     * Flag to see if recovered.
     */
    protected var recovered = false

    /**
     * Handle escalation.
     */
    abstract fun handle()

    /**
     * Get reason.
     *
     * @return reason in textual form.
     */
    abstract fun reason(): String
}
